aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedBackoffStrategies.java

```diff
@@ -103,7 +103,7 @@ public long delayBeforeNextRetry(AmazonWebServiceRequest originalRequest,
                                          AmazonClientException exception,
                                          int retriesAttempted) {
             return (retriesAttempted > MAX_RETRIES) ? maxBackoffTime :
-                    Math.min(((1 << retriesAttempted) * baseDelay), maxBackoffTime);
+                    Math.min(((1L << retriesAttempted) * baseDelay), maxBackoffTime);
```

Comment:
@kiiadi Thanks for your comments,
On 1.11.40 -  MAX_RERTIES is set to 30 , so I did not handle the potential overflow of long, as it does not reach the value of 63.

Using the exponential value  as an int and multiply it by baseDelay might still cause an Integer overflow.

Since the method dleayBeforeNextRetry returns long, I think it's better to resort to longs as soon as possible in the method's code.

I agree that other strategies should be fixed as well.